### DISCUSSION TITLE
jsonGet return

### DISCUSSION THREAD

**Discussion by badminimum** [xetorthio/jedis#3575] [Q&A] [answered]
Hey mates! I wanted to know what will be returned if .jsonGet doesn't exist? Will it be "nil"?

**Comment by sazzad16** [comment]
What do you mean by "if .jsonGet doesn't exist"?

`jsonGet` is a method and it exists in all Jedis versions since 4.0.0 release.

If you `JSON.GET` command, meaning an active Redis server without RedisJSON installed, it would throw a JedisDataException stating unknown command.

If you mean jsonGet method call targeting a non-existing path:
- In RedisJSON v1 you would get `null`.
- In RedisJSON v2 you would get an empty JSONArray (`[]`).

**Reply by badminimum** [reply]
I mean what will be returned if the jsonGet value is nil

**Reply by sazzad16** [reply, marked as answer]
null

**Comment by sazzad16** [comment]
It seems you have many RedisJSON questions. I'd strongly suggest you to do the [RU204 course](https://university.redis.com/courses/ru204/).
